config/v1/oidc/config.proto

```diff
@@ -129,6 +129,24 @@ message OIDCConfig {
     JwksFetcherConfig jwks_fetcher = 17;
   }
 
+  // Available [Client Authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication)
+  // methods.
+  // * `client_secret_basic` - Clients authenticate using the HTTP Basic authentication scheme.
+  // * `client_secret_post` - Clients authenticate by including the Client Credentials in the request body.
+  // * `client_secret_jwt` - Clients create a JWT using an HMAC SHA algorithm, such as HMAC SHA-256 (not implemented).
+  // * `private_key_jwt` - Clients that have registered a public key sign a JWT using that key (not implemented).
+  // * `none` - The Client does not authenticate itself at the Token Endpoint, either because it uses only the
+  //   Implicit Flow (and so does not use the Token Endpoint) or because it is a Public Client with no
+  //   Client Secret or other authentication mechanism (not implemented).
+  // If not set, it defaults to `client_secret_basic`.
+  string client_authentication_method = 25 [(validate.rules).string = {ignore_empty: true, in: [
+    "client_secret_basic", // Clients authenticate using the HTTP Basic authentication scheme.
+    "client_secret_post", // Clients authenticate by including the Client Credentials in the request body.
+    "client_secret_jwt", // Clients create a JWT using an HMAC SHA algorithm, such as HMAC SHA-256.
+    "private_key_jwt", // Clients that have registered a public key sign a JWT using that key.
+    "none" // The Client does not authenticate itself at the Token Endpoint.
+  ]}];
```

Comment:
The problem using enums is that the default protoc linter complains about the naming, but then the UX becomes quite ugly, as the values users would put in the config JSON don't properly reflect the value in the OIDC spec.
Instead, I've opted for just having a string field, and validating the possible values.
This is, I think a nice UX. WDYT?